src/main/java/org/prebid/server/bidder/adtrgtme/AdtrgtmeBidder.java

```diff
@@ -0,0 +1,170 @@
+            return Result.of(bidderBids, errors);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    public MultiMap makeRequestHeaders(Device device) {
+        MultiMap headers = HttpUtil.headers();
+
+        headers.set(HttpUtil.X_OPENRTB_VERSION_HEADER, X_OPENRTB_VERSION);
+        HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.USER_AGENT_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getUa));
+        HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.X_FORWARDED_FOR_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getIpv6));
+        HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.X_FORWARDED_FOR_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getIp));
+
+        return headers;
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse, List<BidderError> errors, List<Imp> imps) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> resolveBidderBid(bidResponse.getCur(), imps, bid, errors))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private BidderBid resolveBidderBid(String currency, List<Imp> imps, Bid bid, List<BidderError> errors) {
+        final BidType bidType;
+        try {
+            bidType = getBidType(bid.getImpid(), imps);
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
+        }
+        return BidderBid.of(bid, bidType, currency);
+    }
+
+    private BidType getBidType(String impId, List<Imp> imps) {
+        for (Imp imp : imps) {
+            if (imp.getId().equals(impId)) {
+                if (imp.getBanner() != null) {
+                    return BidType.banner;
+                } else {
+                    throw new PreBidException(String.format("Unsupported bidtype for bid: \"%s\"", impId));
+                }
+            }
+        }
+        throw new PreBidException("Failed to find impression \"%s\"".formatted(impId));
+    }
+
+    private String resolveRequestUri(List<BidderError> errors, BidRequest singleRequest) {
+        if (singleRequest.getSite() != null) {
+            if (singleRequest.getSite().getId() != null && !singleRequest.getSite().getId().isEmpty()) {
+                return String.format("%s?s=%s&prebid", endpointUrl, singleRequest.getSite().getId());
+            } else {
+                errors.add(BidderError.badInput("request.Site.ID is not provided"));
```

Comment:
Since you have wrapped this method with `try-catch` block which expects PrebidException, this method can just throw without need to pass errors.